src/Plugin/Field/FieldWidget/CommerceNumberWidget.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Drupal\commerce\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\NumberWidget;
+
+/**
+ * Plugin implementation of the 'commerce_number' widget.
+ *
+ * @FieldWidget(
+ *   id = "commerce_number",
+ *   label = @Translation("Commerce number field"),
```

Comment:
Label is a bit ugly. Perhaps we name it "Decimal number field" and prefix the class name accordingly?